packages/react-core/src/demos/examples/AlertGroup/AlertGroupDemo.tsx

```diff
@@ -0,0 +1,423 @@
+    ]);
+
+    if (!isDrawerExpanded) {
+      setAlerts(prevAlerts => [
+        <Alert
+          variant={variant}
+          title={title}
+          timeout={alertTimeout}
+          onTimeout={() => removeAlert(key)}
+          isLiveRegion
+          actionClose={
+            <AlertActionCloseButton title={title} variantLabel={`${variant} alert`} onClose={() => removeAlert(key)} />
+          }
+          key={key}
+          id={key.toString()}
+        >
+          <p>{description}</p>
+        </Alert>,
+        ...prevAlerts
+      ]);
+    }
+  };
+
+  const removeNotification = (key: React.Key) => {
+    setNotifications(prevNotifications => prevNotifications.filter(notification => notification.key !== key));
+  };
+
+  const removeAllNotifications = () => {
+    setNotifications([]);
+  };
+
+  const isNotificationRead = (key: React.Key) =>
+    notifications.find(notification => notification.key === key)?.isNotificationRead === true;
+
+  const markNotificationRead = (key: React.Key) => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification =>
+        notification.key === key ? { ...notification, isNotificationRead: true } : notification
+      )
+    );
+  };
+
+  const markAllNotificationsRead = () => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification => ({ ...notification, isNotificationRead: true }))
+    );
+  };
+
+  const getUnreadNotificationsNumber = () =>
+    notifications.filter(notification => notification.isNotificationRead === false).length;
+
+  const onNotificationBadgeClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(!isDrawerExpanded);
+  };
+
+  const onDropdownToggle = (id: React.Key, isActive: boolean) => {
+    if (!isActive) {
+      setOpenDropdownKey(null);
+      return;
+    }
+    setOpenDropdownKey(id);
+  };
+
+  const onDropdownSelect = () => {
+    setOpenDropdownKey(null);
+  };
+
+  const buildOverflowMessage = () => {
+    const overflow = alerts.length - maxDisplayed;
+    if (overflow > 0 && maxDisplayed > 0) {
+      return `View ${overflow} more notifications in notification drawer`;
+    }
+    return '';
+  };
+
+  const getUniqueId = () => new Date().getTime();
+
+  const getTimeCreated = () => {
+    const date = new Date();
+    return (
+      date.toDateString() +
+      ' at ' +
+      ('00' + date.getHours().toString()).slice(-2) +
+      ':' +
+      ('00' + date.getMinutes().toString()).slice(-2)
+    );
+  };
+
+  const removeAlert = (key: React.Key) => {
+    setAlerts(prevAlerts => prevAlerts.filter(alert => alert.props.id !== key.toString()));
+  };
+
+  const removeAllAlerts = () => {
+    setAlerts([]);
+  };
+
+  const addSuccessAlertNotification = () => {
+    handleNewNotification(
+      'Success alert notification',
+      'Success alert',
+      'success',
+      getUniqueId(),
+      getTimeCreated(),
+      'Success alert notification description'
+    );
+  };
```

Comment:
Nit: Rather than having each of these `addXAlertNotification` functions pass so much similar data, what would you think about refactoring `handleNewNotification` to just accept the variant type for the notification and handling the rest of the notification formatting internally?

Reply:
That would look better definitely :)